### PR TITLE
BAH-923: Adding ability to save specialties 

### PR DESCRIPTION
**Pull request by rubailly**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/appointments/service/SpecialityService.java [hunk 1]
```diff
@@ -15,4 +15,8 @@
     @Transactional
     @Authorized({"View Appointments"})
     List<Speciality> getAllSpecialities();
+    
+    @Transactional
+    @Authorized({"Manage Appointment Services"})
```
Comment (rbuisson) [inline api/src/main/java/org/openmrs/module/appointments/service/SpecialityService.java]: I believe we should create a specific permission for this "Manage Appointment Specialities"

File: api/src/test/java/org/openmrs/module/appointments/service/SpecialityServiceTest.java [hunk 2]
```diff
@@ -58,4 +65,12 @@ public void shouldNotGetAllSpecialitiesIfUserHasNoPrivilege() throws Exception {
         Context.authenticate(noPrivilegeUser, password);
         assertNotNull(specialityService.getAllSpecialities());
     }
+    
+    @Test
+    public void shouldBeAbleToSaveSpecialityIfUserHasManageServicesPrivilege() throws Exception {
```
Comment (rbuisson) [inline api/src/test/java/org/openmrs/module/appointments/service/SpecialityServiceTest.java]: We should also try to save when user does not have the privilege (_Manage Appt. Specialities_) and ensure it fails.

File: api/src/test/java/org/openmrs/module/appointments/service/impl/SpecialityServiceImplTest.java [hunk 3]
```diff
@@ -65,5 +67,13 @@ public void shouldGetAllSpecialities() throws Exception {
         assertEquals(new Integer(1), specialities.get(0).getId());
         assertEquals(creator, specialities.get(0).getCreator());
     }
+    
+    @Test
+    public void testCreateSpeciality() throws Exception {
```
[the remark this answers is not in the document]
Comment (rubailly) [reply api/src/test/java/org/openmrs/module/appointments/service/impl/SpecialityServiceImplTest.java]: Done

File: api/src/main/java/org/openmrs/module/appointments/service/SpecialityService.java (hunk 1 above)
Comment (rubailly) [reply api/src/main/java/org/openmrs/module/appointments/service/SpecialityService.java]: Done

File: api/src/test/java/org/openmrs/module/appointments/service/SpecialityServiceTest.java (hunk 2 above)
Comment (rubailly) [reply api/src/test/java/org/openmrs/module/appointments/service/SpecialityServiceTest.java]: Done

File: api/src/test/java/org/openmrs/module/appointments/service/SpecialityServiceTest.java [hunk 4]
```diff
@@ -58,4 +65,20 @@ public void shouldNotGetAllSpecialitiesIfUserHasNoPrivilege() throws Exception {
...
+        assertNotNull(specialityService.save(speciality));
+    }
+    
+    @Test(expected = APIAuthenticationException.class)
+    public void shouldNotBeAbleToSaveSpecialityIfUserDoesNotHaveManageSpecialitiesPrivilege() throws Exception {
+        Context.authenticate(manageUser, password);
+        Speciality speciality = new Speciality();
+        speciality.setName("speciality");
+        assertNotNull(specialityService.save(speciality));
+    }
```
[the remark this answers is not in the document]
Comment (rubailly) [reply api/src/test/java/org/openmrs/module/appointments/service/SpecialityServiceTest.java]: [Done](https://github.com/Bahmni/openmrs-module-appointments/pull/13/commits/3e22a08ea9be749b200a9c33739b4081bec7f9c1).

**Pull request merged**